MatrixKit/Utils/EventFormatter/MarkdownParser.swift

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2015 OpenMarket Ltd
+Copyright 2017 Vector Creations Ltd
+Copyright 2018 New Vector Ltd
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import Foundation
+import Down
+import libcmark
+
+@objcMembers
+public class MarkdownParser: NSObject {
```

Comment:
In fact this is not a Markdown parser here but a Markdown to HTML renderer. So maybe a class name like `MarkdownToHTMLRenderer` is more appropriate.
To be able to have one renderer instance in `MXKEventFormatter` it can be better to remove the `public init(withMarkdown markdown: String)` method.

In that way we can create a `MarkdownToHTMLRendererProtocol` with this method: 
```
public func renderToHTML(markdown: String) -> String?
```
And use `@property (nonatomic, strong) id<MarkdownToHTMLRendererProtocol> markdownToHTMLRenderer` in `MXKEventFormatter`

Reply:
Updated